src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/HmppsDocumentManagementService.kt

```diff
@@ -45,24 +50,44 @@ class HmppsDocumentManagementService (val hmppsDocumentManagementApiClient: Hmpp
             ?: throw EntityNotFoundException("Document not found /hearing/%s/defendant/%s/documents/%s", hearingId, defendantId, documentId)
     }
 
-    fun getDocument(hearingId: String, defendantId: String, documentId: String): Optional<Mono<ResponseEntity<Flux<InputStreamResource>>>> {
+    fun getDocument(
+        hearingId: String,
+        defendantId: String,
+        documentId: String
+    ): Mono<ResponseEntity<Flux<InputStreamResource>>>? {
```

Comment:
That mono is awfully complex as return type, can we do ResponseEntity<T> or maybe map it?

Reply:
addressed, thanks